src/Models/StatsModel.php

```diff
@@ -46,19 +46,19 @@ public function save($data)
 		// Check if a row exists for this unique ID and update the existing record if so
 		$recordExists = $db->setQuery(
 			$db->getQuery(true)
-				->select('id')
+				->select('unique_id')
 				->from('#__jstats')
 				->where('unique_id = ' . $db->quote($data->unique_id))
 		)->loadResult();
 
 		if ($recordExists)
 		{
-			$data->id = $recordExists;
-			$db->updateObject('#__jstats', $data, ['id']);
+			$data->unique_id = $recordExists;
```

Comment:
This line now becomes unnecessary as the controller sets the unique ID to the data object before injecting it into the method.